src/lang/java/method_body/stmt.rs

```diff
@@ -0,0 +1,374 @@
+use crate::ast::*;
+use crate::java::method_body::parse_block;
+use crate::java::method_body::parse_child_nodes;
+use crate::java::modifier::find_modifier;
+use crate::java::util::vartype::find_type;
+use crate::java::{method_body::log_unknown_tag, modifier::parse_modifiers};
+use crate::ComponentInfo;
+use crate::AST;
+
+use super::{expr::parse_expr, is_common_junk_tag, node::parse_node};
+
+/// File holding all Java statement parsing (e.g., while/for/trycatch)
+
+/// Parse an AST section containing a variable declaration
+pub(crate) fn parse_decl(ast: &AST, component: &ComponentInfo) -> DeclStmt {
+    // Extract informtion about the variable
+    let r#type = find_type(ast);
+    let modifier = find_modifier(ast, &*component.path, &*component.package_name);
+
+    // Determine the value it was set to
+    let rhs = parse_child_nodes(ast, component);
+
+    let mut decl = DeclStmt::new(vec![], vec![]);
+    for var in rhs.iter() {
+        // Extract expression from the hierarchy
+        let base = match var {
+            Node::Stmt(Stmt::ExprStmt(ExprStmt { expr, .. })) | Node::Expr(expr) => expr,
+            _ => {
+                eprintln!("Unable to interpret as variable: {:#?}", var);
+                continue;
+            }
+        };
+
+        // Parse variable
+        match base {
+            Expr::BinaryExpr(expr) => match expr.lhs.as_ref() {
+                Expr::Ident(lhs) => {
+                    decl.variables
+                        .push(VarDecl::new(Some(r#type.clone()), lhs.clone()));
+                    decl.expressions.push(Some(expr.rhs.as_ref().clone()));
+                }
+                unknown => eprintln!("Expected Ident got {:#?}", unknown),
+            },
+            Expr::Ident(id) => {
+                decl.variables
+                    .push(VarDecl::new(Some(r#type.clone()), id.clone()));
+                decl.expressions.push(None);
+            }
+            unknown => {
+                eprintln!("Expected BinaryExpr or Ident, got {:#?}", unknown);
+            }
+        }
+    }
+
+    // TODO: Use name
```

Comment:
Is name still unused?

Reply:
Nope, that's an old warning, thanks for pointing it out.